### PR TITLE
usb: warn about too small REQUEST_BUFFER

### PR DESCRIPTION
**Pull request by jfischer-no**
Warn about too small REQUEST_BUFFER.
Clarify usb_handle_request function parameter.

### CODE REVIEW

File: subsys/usb/usb_device.c [hunk 1]
```diff
@@ -232,9 +232,11 @@ static bool usb_handle_request(struct usb_setup_packet *setup,
 /*
  * @brief send next chunk of data (possibly 0 bytes) to host
  *
+ * @param [in] rlen        Requested length
+ *
  * @return N/A
  */
-static void usb_data_to_host(u16_t len)
+static void usb_data_to_host(u16_t rlen)
```
Comment (finikorg) [inline subsys/usb/usb_device.c]: This does not belong to this patch and not needed IMO

Comment (jfischer-phytec-iot) [reply subsys/usb/usb_device.c]: I will move it to another PR. But it is necessary. There are many `len`, can you say without much thought what is meant by `len` in the context?

Comment (finikorg) [reply subsys/usb/usb_device.c]: At least I know that it is len, rlen is not better here.

File: subsys/usb/usb_device.c [hunk 2]
```diff
@@ -332,6 +332,10 @@ static void usb_handle_control_transfer(u8_t ep,
 
 		/* Send smallest of requested and offered length */
 		usb_dev.data_buf_residue = MIN(usb_dev.data_buf_len, length);
+		if (usb_dev.data_buf_len > CONFIG_USB_REQUEST_BUFFER_SIZE) {
+			LOG_ERR("REQUEST_BUFFER too small");
+		}
```
Comment (carlescufi) [inline subsys/usb/usb_device.c]: Can we add a call to `k_oops()` here so that the error is unconditionally detected

Comment (jfischer-phytec-iot) [reply subsys/usb/usb_device.c]: done

File: subsys/usb/usb_device.c [hunk 3]
```diff
@@ -332,6 +332,11 @@ static void usb_handle_control_transfer(u8_t ep,
 
 		/* Send smallest of requested and offered length */
 		usb_dev.data_buf_residue = MIN(usb_dev.data_buf_len, length);
+		if (usb_dev.data_buf_len > CONFIG_USB_REQUEST_BUFFER_SIZE) {
+			LOG_ERR("REQUEST_BUFFER too small");
```
Comment (emob-nordic) [inline subsys/usb/usb_device.c]: I the same function above
https://github.com/zephyrproject-rtos/zephyr/blob/e6dc03550cd934e18a6748f437483be82f14c4b9/subsys/usb/usb_device.c#L306
there is similar logging based on wrong buffer length. It's done based on 'Data transfer direction' but I am wondering if its possible to concatenate this into one LOG_ERR to make it more clear? 

What is more. Is it really an error? Refering to 9.3.5 of usb 2.0 spec:

> On an input request, a device must never return more data than is indicated by the wLength value; it may
return less. On an output request, wLength will always indicate the exact amount of data to be sent by the
host. Device behavior is undefined if the host should send more data than is specified in wLength.

File: samples/subsys/usb/hid-cdc/prj.conf [hunk 4]
```diff
@@ -14,6 +14,7 @@ CONFIG_USB_DEVICE_LOG_LEVEL_INF=y
 
 CONFIG_USB_CDC_ACM=y
 CONFIG_USB_CDC_ACM_DEVICE_COUNT=2
+CONFIG_USB_REQUEST_BUFFER_SIZE=192
```
Comment (emob-nordic) [inline samples/subsys/usb/hid-cdc/prj.conf]: Why is size of 192 needed?

File: samples/subsys/usb/cdc_acm_composite/prj.conf [hunk 5]
```diff
@@ -12,6 +12,7 @@ CONFIG_USB_COMPOSITE_DEVICE=y
 CONFIG_USB_CDC_ACM=y
 CONFIG_USB_CDC_ACM_DEVICE_COUNT=2
 CONFIG_USB_CDC_ACM_RINGBUF_SIZE=512
+CONFIG_USB_REQUEST_BUFFER_SIZE=192
```
Comment (emob-nordic) [inline samples/subsys/usb/cdc_acm_composite/prj.conf]: Why is size of 192 needed?

**Comment by jfischer-phytec-iot** [comment]
It is not quite right, I have overlooked some passages. In general it is difficult to control what happens with `usb_dev.data_buf `and `usb_dev.req_data`.
